test/anti_patterns/long_parameter_list_test.dart

```diff
@@ -0,0 +1,49 @@
+@TestOn('vm')
+import 'package:analyzer/dart/analysis/features.dart';
+import 'package:analyzer/dart/analysis/utilities.dart';
+import 'package:dart_code_metrics/src/anti_patterns/long_parameter_list.dart';
+import 'package:dart_code_metrics/src/models/config.dart';
+import 'package:dart_code_metrics/src/models/source.dart';
+import 'package:test/test.dart';
+
+const _content = '''
+
+void main() {}
+
+void func(int a, int b, int c, int e) {}
+
+void func2(int a, int b, int c, int e, String f) {}
+
+''';
+
+void main() {
+  test('LongParameterList report about found design issues', () {
+    final sourceUrl = Uri.parse('/example.dart');
+
+    final parseResult = parseString(
+        content: _content,
+        featureSet: FeatureSet.fromEnableFlags([]),
+        throwIfDiagnostics: false);
+
+    final issues = LongParameterList().check(
+        Source(sourceUrl, parseResult.content, parseResult.unit),
+        const Config());
+
+    expect(issues.length, equals(1));
+
+    expect(issues.single.patternId, equals('long-parameter-list'));
+    expect(issues.single.patternDocumentation.toString(),
+        equals('https://git.io/JUGrU'));
+    expect(issues.single.sourceSpan.sourceUrl, equals(sourceUrl));
+    expect(issues.single.sourceSpan.start.offset, equals(59));
+    expect(issues.single.sourceSpan.start.line, equals(6));
+    expect(issues.single.sourceSpan.start.column, equals(1));
+    expect(issues.single.sourceSpan.end.offset, equals(110));
+    expect(issues.single.message,
```

Comment:
What do you think about moving `issues.single` to separate variable? And maybe `issues.single.sourceSpan` too?